target_chains/ethereum/contracts/forge-test/Entropy.t.sol

```diff
@@ -1822,57 +1836,14 @@ contract EntropyTest is Test, EntropyTestUtils, EntropyEvents, EntropyEventsV2 {
 
             assertEq(entries.length, 2);
             // first entry is CallbackFailed which we aren't going to check.
-            // Unfortunately event.selector was added in Solidity 0.8.15 and we're on 0.8.4 so we have to copy this spec here.
-            assertEq(
-                entries[1].topics[0],
-                keccak256(
-                    "Revealed(address,address,uint64,bytes32,bool,bytes,uint32,bytes)"
-                )
-            );
-            // Verify the topics match the expected values
-            assertEq(
-                entries[1].topics[1],
-                bytes32(uint256(uint160(provider1)))
-            );
-            assertEq(
-                entries[1].topics[2],
-                bytes32(uint256(uint160(address(consumer))))
-            );
-            assertEq(entries[1].topics[3], bytes32(uint256(sequenceNumber)));
-
-            // Verify the data field contains the expected values (per event ABI)
-            (
-                bytes32 randomNumber,
-                bool callbackFailed,
-                bytes memory callbackErrorCode,
-                uint32 callbackGasUsed,
-                bytes memory extraArgs
-            ) = abi.decode(
-                    entries[1].data,
-                    (bytes32, bool, bytes, uint32, bytes)
-                );
-
-            assertEq(
-                randomNumber,
-                random.combineRandomValues(
-                    userRandomNumber,
-                    provider1Proofs[sequenceNumber],
-                    0
-                )
-            );
-            assertEq(callbackFailed, true);
-            assertEq(callbackErrorCode, bytes(""));
-
-            // callback gas usage is approximate
-            assertTrue(
-                random.getProviderInfoV2(provider1).defaultGasLimit == 0 ||
-                    ((callbackGasUsage * 90) / 100 < callbackGasUsed)
-            );
-            assertTrue(
-                random.getProviderInfoV2(provider1).defaultGasLimit == 0 ||
-                    (callbackGasUsed < (callbackGasUsage * 110) / 100)
+            assertRevealedEvent(
```

Comment:
extracted the test out to a method. the factoring could be a bit odd because i'm working around the solidity "stack too deep" error